src/common_interface/solve.jl

```diff
@@ -593,53 +603,56 @@ function DiffEqBase.__init(
     end
 
     if prob.f.mass_matrix != LinearAlgebra.I
-        if MassLinearSolver == :Dense
+        if LinearSolver in (:Dense, :LapackDense)
+            nojacobian = false
             M = SUNDenseMatrix(length(u0),length(u0))
-            MLS = SUNLinSol_Dense(u0,M)
-            ARKStepSetMassLinearSolver(mem,MLS,M,false)
             _M = MatrixHandle(M,DenseMatrix())
-            _MLS = LinSolHandle(MLS,Dense())
-        elseif MassLinearSolver == :Band
-            M = SUNBandMatrix(length(u0), alg.mass_upper, alg.mass_lower)
-            MLS = SUNLinSol_Band(u0,M)
-            ARKStepSetMassLinearSolver(mem,MLS,M,false)
+            if LinearSolver === :Dense
+                MLS = SUNLinSol_Dense(u0,M)
+                _MLS = LinSolHandle(LS,Dense())
```

Comment:
Should be `MLS` - that's what's causing the CI failures.
That said, This code is horribly messy, and I would recommend cleaning it up to DRY it up to avoid errors caused by this kind of copy paste thing.

Reply:
(same issue in the three instances below)

Reply:
Yeah, the code clearly got one new feature a month until it's the way it is. Someone may need to sit down sometime and refactor it, but it comes up rarely enough that I'd rather work to get pure-Julia stuff than actually do this. But for anyone reading, I would appreciate some cleanup here.

Reply:
I think maybe put in the README about help needed and open an issue?

Reply:
Or advertise heavily this issue for October